test/common/schemaform/ArrayField.unit.spec.jsx

```diff
@@ -166,5 +169,31 @@ describe('Schemaform ArrayField', () => {
       expect(onChange.firstCall.args[0].length).to.equal(3);
       expect(tree.getMountedInstance().state.editing[2]).to.be.false;
     });
+    it('add when invalid', () => {
+      errorSchema[1] = { __errors: ['Test error'] };
+      tree.getMountedInstance().handleAdd();
+
+      expect(tree.getMountedInstance().state.touchedSchema[1]).to.be.true;
+    });
+    it('remove', () => {
+      expect(tree.everySubTree('SchemaField').length).to.equal(1);
```

Comment:
In this case, no. Normally the onChange prop would end up triggering another render with new props, but we're using a spy for that, so nothing changes. We could mock up the onChange prop to force a re-render the component with updated formData, but that seemed like overkill.